nni/nas/hub/pytorch/autoformer.py

```diff
@@ -398,14 +398,65 @@ def __init__(
     def get_extra_mutation_hooks(cls):
         return [MixedAbsPosEmbed.mutate, MixedClsToken.mutate]
 
+    @classmethod
+    def preset(cls, name: str):
+        """Get the model space config proposed in paper."""
+        name = name.lower()
+        assert name in ['tiny', 'small', 'base']
+        init_kwargs = {'qkv_bias': True, 'drop_rate': 0.0, 'drop_path_rate': 0.1, 'global_pool': True, 'num_classes': 1000}
+        if name == 'tiny':
+            init_kwargs.update({
+                'search_embed_dim': (192, 216, 240),
+                'search_mlp_ratio': (3.0, 3.5, 4.0),
+                'search_num_heads': (3, 4),
+                'search_depth': (12, 13, 14),
+            })
+        elif name == 'small':
+            init_kwargs.update({
+                'search_embed_dim': (320, 384, 448),
+                'search_mlp_ratio': (3.0, 3.5, 4.0),
+                'search_num_heads': (5, 6, 7),
+                'search_depth': (12, 13, 14),
+            })
+        elif name == 'base':
+            init_kwargs.update({
+                'search_embed_dim': (528, 576, 624),
+                'search_mlp_ratio': (3.0, 3.5, 4.0),
+                'search_num_heads': (8, 9, 10),
+                'search_depth': (14, 15, 16),
+            })
+        else:
+            raise ValueError(f'Unsupported architecture with name: {name}')
+        return init_kwargs
+
+    @classmethod
+    def load_strategy_checkpoint(cls, name: str, download: bool = True, progress: bool = True):
+        legal = ['random-one-shot-tiny', 'random-one-shot-small', 'random-one-shot-base']
+        if name not in legal:
+            raise ValueError(f'Unsupported name: {name}. It should be one of {legal}.')
+        name = name[16:]
+        from nni.nas.strategy import RandomOneShot
+        init_kwargs = cls.preset(name)
+        model_sapce = cls(**init_kwargs)
+        strategy = RandomOneShot(mutation_hooks=cls.get_extra_mutation_hooks())
+        strategy.attach_model(model_sapce)
+        weight_file = load_pretrained_weight(f"autoformer-{name}-supernet", download=download, progress=progress)
+        pretrained_weights = torch.load(weight_file)
+        assert strategy.model is not None
+        strategy.model.load_state_dict(pretrained_weights)
+        return strategy
+
     @classmethod
     def load_searched_model(
         cls, name: str,
         pretrained: bool = False, download: bool = False, progress: bool = True
     ) -> nn.Module:
-
-        init_kwargs = {'qkv_bias': True, 'drop_rate': 0.0, 'drop_path_rate': 0.1, 'global_pool': True, 'num_classes': 1000}
-        if name == 'autoformer-tiny':
+        legal = ['autoformer-tiny', 'autoformer-small', 'autoformer-base']
+        if name not in legal:
+            raise ValueError(f'Unsupported name: {name}. It should be one of {legal}.')
+        name = name[11:]
```

Comment:
I am a little confused, you use the name "autoformer-xxxx", then slice the name to be "xxxx".

Reply:
wow, still working after hours -__-.
This is because the previous version used the format `autoformer- XXX`. So just to be consistent and compatible, we're going to keep it that way.

Reply:
haha, I am staying at home all day, thus, always working time...